### ISSUE TITLE
Cannot favorite (share) items owned by others with `shareItemWithGroup` as an organizational user

### ISSUE THREAD

**Issue by skitterm** [Esri/arcgis-rest-js#454]
I want to favorite living atlas content (which is not owned by me) using `arcgis-rest-sharing`'s  `shareItemWithGroup`. I can do so with a public account, but not with an organizational account.

The error I get is:
`{"error":{"code":403,"messageCode":"GWM_0003","message":"You do not have permissions to access this resource or perform this operation.","details":[]}}`

I am invoking like this, passing the item owner:
```js
shareItemWithGroup({
      id: 'item-id',
      groupId: user.favGroupId,
      authentication: userSession,
      owner: 'owner-of-item',
    })
```

With public account (it succeeds), the REST request URL is in this format: `https://www.arcgis.com/sharing/rest/content/items/item-id/share`

With organizational account (it fails), the URL is like this: `https://my-org.maps.arcgis.com/sharing/rest/content/users/owner-of-item/items/item-id/share`

The token is passed in both requests. I am able to share my own (and members of my organization's) items.

Not sure why `/users` is being used in some cases and `/items` in others; could that be why the organizational request is failing?

I can put in a repro case if needed.

**Comment by jgravois** [comment]
the error you're reporting is limited to scenarios in which an organizational user with **admin privileges** attempts to update the group sharing of a public item owned a user in an entirely different org.

https://github.com/Esri/arcgis-rest-js/blob/8be88a399a35e1dc31f77f377fb095489e055b2f/packages/arcgis-rest-sharing/src/group-sharing.ts#L110-L115

@dbouwman besides consistency with the docs, is there a tangible benefit to organizational admins using [Share item as item owner](https://developers.arcgis.com/rest/users-groups-and-items/share-item-as-item-owner-.htm)?

if there is, we could introduce something like `outsideOrg: boolean` as _yet another_ input parameter, but my instinct is that we're better off just directing org admins to [Share item as group admin](https://developers.arcgis.com/rest/users-groups-and-items/share-item-as-group-admin-.htm) instead.

**Comment by jgravois** [comment]
i've proposed a fix in #456 that doesn't alter the public facing API.
